### ISSUE TITLE
Albion Online cannot be found in GNOME Software

### ISSUE THREAD

**Issue by AsciiWolf** [flathub/com.albiononline.AlbionOnline#3]
I have flathub repo added, searching for _Albion Online_ in GNOME Software returns no results.

**Comment by AsciiWolf** [comment]
![gnome-software_albion](https://user-images.githubusercontent.com/2339930/27743523-947eaf58-5dbc-11e7-9de7-d635f468243a.png)

```
[asciiwolf@desktop ~]$ flatpak remote-ls --user flathub|grep Albion
com.albiononline.AlbionOnline
```

**Comment by TingPing** [comment]
By default I believe gnome-software hides proprietary apps, you can toggle that in `dconf-editor`.

**Comment by AsciiWolf** [comment]
@TingPing I don't think that's the case. I can see Steam and other proprietary apps without any problem.

**Comment by valentindavid** [comment]
Because it is installed. Uninstall Steam and fully restart gnome-software. There is a background process to kill to fully restart. Then search for Steam.

I have tried with Spotify. It cannot be listed if not installed.

**Comment by valentindavid** [comment]
However after, enabling "/org/gnome/software/show-non-free-software", I still do not see non install proprietary apps.

**Comment by valentindavid** [comment]
Actually Steam shows up. However Albion Online, Discord and Spotify do not.

**Comment by valentindavid** [comment]
GeoGebra also shows up. The main difference I see in the appstream is the presence or not of icon.

**Comment by valentindavid** [comment]
It is on purpose that GNOME Software does not show applications without icon. We do not have any icon for Albion Online on the appstream, because the icon is in the "extra" data downloaded from Albion Online's server at installation. I am not sure what would be the legality of hosting the icon file directly in Flathub.

**Comment by TingPing** [comment]
Ah, if they don't allow redistribution of their icons then I don't think we should. Maybe @hughsie knows of an alternative?

**Comment by hughsie** [comment]
You can have a 'remote' icon in the appdata file that is downloaded in the client. Thus us how we do Webapp like Twitter where the icon can't be shipped.

**Comment by valentindavid** [comment]
Thanks for the info.

I am not sure however that flatpak can generate remote icons in the appstream. I cannot give it in the appdata, icons are not allowed there.

**Comment by hughsie** [comment]
Is it a fatal validation error? I've never tried using a remote icon for a desktop app, normally the icon is redistributable.

**Comment by valentindavid** [comment]
"appstream-util validate-relax" refuses it.

According to the specification it is for appstream collection. Not for upstream metadata (appdata).

**Comment by hughsie** [comment]
Sure, but (as the author of appstream-util) sometimes we need to be pragmatic.

**Comment by valentindavid** [comment]
I think flatpak calls "appstream-compose" on the result of the build.

In my case I have an appdata XML file. But no desktop file yet, nor icons.

If this command is fixed, maybe it would work for me.

**Comment by AsciiWolf** [comment]
What about using some generic placeholder icon for the AppStream metadata?

**Comment by valentindavid** [comment]
You mean using a generic icon in a temporary .desktop file? The appstream icon comes from the .desktop file, not the metadata.

**Comment by AsciiWolf** [comment]
@valentindavid 

>You mean using a generic icon in a temporary .desktop file?

Yes.

**Comment by andrewschott** [comment]
Has anyone asked them?  I have found a pleasant question regarding limited permission to do things goes a long way.

Just asking.

**Comment by AsciiWolf** [comment]
Sandbox Interactive employee told me that it is probably okay to use an icon from the Albion Online [press kit](https://forum.albiononline.com/index.php/Thread/45665-Albion-Online-fan-guild-sites-graphic-resources/?postID=462843#post462843), but he was not 100% sure.

**Comment by TingPing** [comment]
Press kit is a good start, but I don't see any actual legal documents over theirs?

**Comment by AsciiWolf** [comment]
I wrote to the company's CEO about this but did not get any reply yet.

**Comment by TingPing** [comment]
The remote image issue was fixed recently: https://github.com/hughsie/appstream-glib/commit/74ebf5192296f0ef26f6c51ee5f103d9fd72a3df

So once that is released and lands on the builders we can use remote images.

**Comment by AsciiWolf** [comment]
Great!

**Comment by AsciiWolf** [comment]
@TingPing Any news regarding the remote images?

**Comment by TingPing** [comment]
Yes it is supported now, see the other proprietary apps like Skype, Discord, etc for examples.

**Comment by AsciiWolf** [comment]
Why is it not supported yet in Albion Online?

**Comment by valentindavid** [comment]
Do you have an URL to an icon?

**Comment by AsciiWolf** [comment]
What about [this one](https://albiononline.com/wp-content/uploads/2013/07/ipad_icon.png)?

**Comment by AsciiWolf** [comment]
Although I see that the icon is probably in a bad resolution.

**Comment by hughsie** [comment]
64x64 is the minimum size icon that it will use. R

**Comment by nedrichards** [comment]
This is also required to use icons on the upcoming [apps page](https://beta.flathub.org/apps) (still in beta). However, we can now host these icons if we wish - you don't have to use a remote icon if you don't want to.

**Comment by AsciiWolf** [comment]
This issue was fixed in #16.
